torch/csrc/autograd/profiler_python.cpp

```diff
@@ -938,6 +975,20 @@ class PostProcess {
         pop(i.second, end_time_);
       }
     }
+
+    // Assign system TIDs to start events based on the system TID of the next
+    // observed event with the same Python TID.
+    ska::flat_hash_map<size_t, size_t> tid_map;
+    auto it = out.rbegin();
+    for (C10_UNUSED auto _ : c10::irange(initial_size, out.size())) {
+      const auto python_tid =
+          c10::get<ExtraFields<E>>((*it)->extra_fields_).python_tid_;
```

Comment:
if you want to be really paranoid, Python thread ids are just results of `pthread_self` on linux: https://github.com/python/cpython/blob/2f2fa03ff3d566b675020787e23de8fb4ca78e99/Python/thread_pthread.h#L313 (and native_thread_id was added in 3.11 only). So they technically could be reused between invocations.

If you did bookkeeping on RecordFunction::currentThreadId <-> pthread_self level I guess it'd be safer (as you'd do lookups only while the thread is active), but it might be an overkill.